hips/draw/simple.py

```diff
@@ -65,12 +30,64 @@ class SimpleTilePainter:
         HiPS survey properties
     tile : `HipsTile`
        An object of HipsTile
+    tile_format : `str`
+        Format of HiPS tile
     """
 
-    def __init__(self, geometry: WCSGeometry, hips_survey: HipsSurveyProperties, tile: HipsTile) -> None:
+    def __init__(self, geometry: WCSGeometry, hips_survey: HipsSurveyProperties, tile_format: str) -> None:
         self.geometry = geometry
         self.hips_survey = hips_survey
-        self.tile = tile
+        self.tile_format = tile_format
+
+    @property
+    def hips_order(self) -> int:
+        """Compute HiPS tile order matching a given image pixel size.
+
+        Examples
+        --------
+        >>> from hips.utils import WCSGeometry
+        >>> from hips.draw import SimpleTilePainter
+        >>> from hips.tiles import HipsSurveyProperties
+        >>> from astropy.coordinates import SkyCoord
+        >>> url = 'http://alasky.unistra.fr/DSS/DSS2Merged/properties'
+        >>> hips_survey = HipsSurveyProperties.fetch(url)
+        >>> geometry = WCSGeometry.create_simple(
+        ...     skydir=SkyCoord(0, 0, unit='deg', frame='icrs'),
+        ...     width=2000, height=1000, fov="3 deg",
+        ...     coordsys='icrs', projection='AIT'
+        ... )
+        >>> simple_tile_painter = SimpleTilePainter(geometry, hips_survey, 'fits')
+        >>> simple_tile_painter.hips_order
+        7
+        """
+        # Sky image angular resolution (pixel size in degree)
+        resolution = np.min(proj_plane_pixel_scales(self.geometry.wcs))
+        desired_order = self._get_hips_order_for_resolution(self.hips_survey.tile_width, resolution)
+        # Return the desired order, or the highest resolution available.
+        # Note that HiPS never has resolution less than 3,
+        # and that limit is handled in _get_hips_order_for_resolution
+        return np.min([desired_order, self.hips_survey.hips_order])
+
+    def _get_hips_order_for_resolution(self, tile_width, resolution):
+        """Finding the best HiPS order by looping through all possible options."""
+        tile_order = np.log2(tile_width)
+        full_sphere_area = 4 * np.pi * np.square(180 / np.pi)
+        # 29 is the maximum order supported by healpy and 3 is the minimum order
+        for candidate_tile_order in range(3, 29 + 1):
+            tile_resolution = np.sqrt(full_sphere_area / 12 / 4 ** (candidate_tile_order + tile_order))
+            # Finding the smaller tile order with a resolution equal or better than geometric resolution
+            if tile_resolution <= resolution:
+                break
+
+        return candidate_tile_order
+
+    @property
+    def tile_indices(self):
```

Comment:
Please always add at least one-line docstrings to every property, function, class, method, ... everything that appears in the public API docs. The most important thing to mention is what it is and what type it is.